swarm.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"path"
+	"runtime"
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+func deploySwarmStack(logger *zap.SugaredLogger, cmd SwarmDeployCommand, clonePath string) error {
+	command := getDockerBinaryPath()
+	args := make([]string, 0)
+
+	if cmd.Prune {
+		args = append(args, "stack", "deploy", "--prune", "--with-registry-auth")
+	} else {
+		args = append(args, "stack", "deploy", "--with-registry-auth")
+	}
+	if !cmd.Pull {
+		args = append(args, "--resolve-image=never")
+	}
+
+	for _, cfile := range cmd.ComposeRelativeFilePaths {
+		args = append(args, "--compose-file", path.Join(clonePath, cfile))
+	}
+	logger.Infow("Deploying Swarm stack", "composeFilePaths", cmd.ComposeRelativeFilePaths,
+		"workingDirectory", clonePath, "projectName", cmd.ProjectName)
+	args = append(args, cmd.ProjectName)
+
+	err := runCommandAndCaptureStdErr(command, args, cmd.Env, clonePath)
+	if err != nil {
+		logger.Errorw("Failed to swarm deploy Git repository", "error", err)
+		return errDeployComposeFailure
+	}
+	logger.Info("Swarm stack deployment complete")
+
+	return err
+}
+
+func checkRunningService(logger *zap.SugaredLogger, cmd SwarmDeployCommand) ([]string, error) {
```

Comment:
the cmd is only needed for the project name, I suggest just passing the project name, it makes it clearer